ksqldb-engine/src/main/java/io/confluent/ksql/engine/SandboxedExecutionContext.java

```diff
@@ -127,11 +127,13 @@ public ExecuteResult execute(
       final ServiceContext serviceContext,
       final ConfiguredKsqlPlan ksqlPlan
   ) {
-    return EngineExecutor.create(
+    final ExecuteResult result = EngineExecutor.create(
         engineContext,
         serviceContext,
         ksqlPlan.getConfig()
     ).execute(ksqlPlan.getPlan());
+    result.getQuery().ifPresent(query -> query.getKafkaStreams().close());
```

Comment:
Why does it close it during the execution?

Reply:
It's so we don't have to close the KafkaStreams app in all the various locations where we use the sandbox execution.